Keas.Mvc/ClientApp/components/Spaces/SpacesContainer.tsx

```diff
@@ -125,38 +125,53 @@ export default class SpacesContainer extends React.Component<{}, IState> {
         this.context.router.history.push(`${this._getBaseUrl()}/spaces`);
     };
 
-    private _returnToSpaceDetails = (spaceId: number) => {
-        this.context.router.history.push(`${this._getBaseUrl()}/spaces/details/${spaceId}`);
-    }
-
-    private _workstationAssigned = (spaceId: number, created: boolean, assigned: boolean) => {
+      // managing counts
+    private _assetAssignedOrCreated = (type: string, spaceId: number, personId: number, created: boolean, assigned: boolean) => {
         const index = this.state.spaces.findIndex(x => x.id === spaceId);
         if(index > -1)
         {
             const spaces = [...this.state.spaces];
-            if(created)
-            {
-                spaces[index].workstationsTotal++;
-            }
-            if(assigned)
-            {
-                spaces[index].workstationsInUse++;
+            switch(type) {
+            case "equipment": 
+                spaces[index].equipmentCount++;
+                break;
+            case "key":
+                spaces[index].keyCount++;
+                break;
+            case "workstation":
+                if(created) {
+                    spaces[index].workstationsTotal++;
+                }
+                if(assigned)
+                {
+                    spaces[index].workstationsInUse++;
+                }
             }
             this.setState({spaces});
         } 
     }
 
-    private _workstationRevoked = (spaceId: number) => {
-        const index = this.state.spaces.findIndex(x => x.id === spaceId);
-        if(index > -1)
-        {
-            const spaces = [...this.state.spaces];
+    private _assetRevokedOrDeleted = (type: string, spaceId: number, personId: number) => {
+    const index = this.state.spaces.findIndex(x => x.id === spaceId);
```

Comment:
@laholstege seems like this code is pretty common across multiple files,  Might not be worth it but think about moving to a common util method you can just import into the page.  Something where you'd pass the list, the index you care about, the delta, and the type and let the method figure it out.